ember-core/src/main/scala/org/http4s/ember/core/Util.scala

```diff
@@ -17,12 +17,47 @@
 package org.http4s.ember.core
 
 import scala.concurrent.duration._
+import org.http4s.HttpVersion
+import org.http4s.Headers
+import org.http4s.Header
+import org.http4s.headers.Connection
+import org.typelevel.ci._
+import cats.data.NonEmptyList
+import java.util.Locale
 
 private[ember] object Util {
 
+  private[this] val closeCi = ci"close"
+  private[this] val keepAliveCi = ci"keep-alive"
+  private[this] val connectionCi = ci"connection"
+  private[this] val close = Connection(NonEmptyList.of(closeCi))
+  private[this] val keepAlive = Connection(NonEmptyList.one(keepAliveCi))
+
   def durationToFinite(duration: Duration): Option[FiniteDuration] = duration match {
     case f: FiniteDuration => Some(f)
     case _ => None
   }
 
+  def connectionFor(httpVersion: HttpVersion, headers: Headers): Connection =
+    if (isKeepAlive(httpVersion, headers)) keepAlive
+    else close
+
+  def isKeepAlive(httpVersion: HttpVersion, headers: Headers): Boolean = {
+    // We know this is raw because we have not parsed any headers in the underlying alg.
+    // If Headers are being parsed into processed for in ParseHeaders this is incorrect.
+    // TODO: the problem is that any string that contains `expected` is admissible
+    def hasConnection(expected: String): Boolean =
+      headers.headers.exists {
+        case Header.Raw(name, value) =>
+          name == connectionCi && value.toLowerCase(Locale.ROOT).contains(expected)
+        case _ => false
+      }
+
+    httpVersion match {
+      case HttpVersion.`HTTP/1.0` => hasConnection(keepAliveCi.toString)
+      case HttpVersion.`HTTP/1.1` => !hasConnection(closeCi.toString)
+      case _ => false
```

Comment:
This feels like a time bomb in a future HTTP version, but I guess those probably won't go through this code path.

Reply:
Could go with the `sys.error()` trick so that we at least get a runtime error rather than some silent, obscure behavior